src/basic/fd-util.c

```diff
@@ -187,22 +187,21 @@ int fd_cloexec(int fd, bool cloexec) {
 }
 
 int fd_cloexec_many(const int fds[], size_t n_fds, bool cloexec) {
-        int ret = 0, r;
+        int r = 0;
 
         assert(fds || n_fds == 0);
 
         FOREACH_ARRAY(fd, fds, n_fds) {
                 if (*fd < 0) /* Skip gracefully over already invalidated fds */
                         continue;
 
-                r = fd_cloexec(*fd, cloexec);
-                if (r < 0) /* Continue going, but return first error */
-                        RET_GATHER(ret, r);
-                else
-                        ret = 1; /* report if we did anything */
+                RET_GATHER(r, fd_cloexec(*fd, cloexec));
+
+                if (r >= 0)
+                        r = 1; /* report if we did anything */
```

Comment:
Then, I think this should be moved at the end of the function, as the next iteration may be failed.
```c
FOREACH_ARRAY(...) {
        RET_GATHER(r, ...);
}

return r < 0 ? r : 1; /* report if we did anything. */
```
 But, not a big matter.

Reply:
Although very unusual, that would ignore the case where all passed fds are negative.

Or maybe we should only accept positive fds?

Reply:
Ah, you are right. I should take coffee.

Reply:
Don't stay up too late:)